src/server/frames.ts

```diff
@@ -981,6 +981,24 @@ export class Frame extends SdkObject {
     }, this._page._timeoutSettings.timeout(options));
   }
 
+  async dragAndDrop(metadata: CallMetadata, selector1: string, selector2: string,  options: types.PointerActionWaitOptions & types.NavigatingActionWaitOptions = {}) {
+    const controller = new ProgressController(metadata, this);
+    await controller.run(async progress => {
+      await dom.assertDone(await this._retryWithProgressIfNotConnected(progress, selector1, async handle => {
+        return handle._retryPointerAction(progress, 'move and down', false, async point => {
+          await this._page.mouse.move(point.x, point.y);
+          await this._page.mouse.down();
+        }, options);
+      }));
+      await dom.assertDone(await this._retryWithProgressIfNotConnected(progress, selector2, async handle => {
+        return handle._retryPointerAction(progress, 'move and up', false, async point => {
+          await this._page.mouse.move(point.x, point.y);
+          await this._page.mouse.up();
+        }, options);
```

Comment:
You need to convert timeout to deadline before calling the first part and then convert deadline into timeout and pass it into the remainder here.